### PR TITLE
spike/config per environment

### PR DESCRIPTION
**Pull request by feba-rajan**
## Data Template

**Ticket**
- Link: https://github.com/orgs/digital-land/projects/9/views/5?pane=issue&itemId=84357146&issue=digital-land%7Ctechnical-documentation%7C126

### CODE REVIEW

File: .github/workflows/upload_to_S3.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.DEPLOY_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.DEPLOY_AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      - name: Save to S3
+        run: |
+          aws s3 sync ./collection s3://${{ secrets.DEPLOY_COLLECTION_DATA_BUCKET }}/config/collection --delete
```
Comment (eveleighoj) [inline .github/workflows/upload_to_S3.yml:50]: We don't want to delete at this point, it should automatically override the files but there could be files that exist there at the minute that aren't in the repo

File: .github/workflows/upload_to_S3.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: Upload to S3
+
+on:
+  push:
+    branches:
+      - spike/config-per-environment
```
Comment (eveleighoj) [inline .github/workflows/upload_to_S3.yml:6]: This probably needs to be main

File: .github/workflows/upload_to_S3.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          aws-secret-access-key: ${{ secrets.DEPLOY_AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      - name: Save to S3-Development
+        if: ${{ matrix.environment == 'Development' }}
+        run: |
+          aws s3 sync ./collection s3://${{ secrets.DEPLOY_COLLECTION_DATA_BUCKET }}/config/collection
+          aws s3 sync ./pipeline s3://${{ secrets.DEPLOY_COLLECTION_DATA_BUCKET }}/config/pipeline
+
+      - name: Configure AWS Credentials - Staging
```
Comment (cpcundill) [inline .github/workflows/upload_to_S3.yml:55]: A single upload to S3 is all that is required.  Just rely on the environment variables set from the environment.  This job will be invoked for each environment anyway so we don't need to explicitly upload to dev, staging, prod

File: .github/workflows/upload_to_S3.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+  
+      - name: Save to S3 - Staging
+        if: ${{ matrix.environment == 'Staging' }}
+        run: |
+          aws s3 sync ./collection s3://${{ secrets.STAGING_COLLECTION_DATA_BUCKET }}/config/collection
+          aws s3 sync ./pipeline s3://${{ secrets.STAGING_COLLECTION_DATA_BUCKET }}/config/pipeline
+  
+      - name: Configure AWS Credentials - Production
```
Comment (cpcundill) [inline .github/workflows/upload_to_S3.yml:69]: Same as per comment above: a single upload to S3 is all that is required. Just rely on the environment variables set from the environment. This job will be invoked for each environment anyway so we don't need to explicitly upload to dev, staging, prod

File: .github/workflows/upload_to_S3.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        uses: actions/checkout@v4
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.DEPLOY_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.DEPLOY_AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      # - name: Save to S3
```
Comment (cpcundill) [inline .github/workflows/upload_to_S3.yml:48]: We're not publishing to S3 anymore with this commented?  Is this intentional?

**Review by cpcundill — CHANGES_REQUESTED** [review]

File: .github/workflows/upload_to_S3.yml (hunk 5 above)
Comment (feba-rajan) [reply .github/workflows/upload_to_S3.yml:48]: Yes, commenting out the S3 sync lines is intentional for testing purposes, the goal is to merge this into main, and then create a new branch to uncomment the lines.

**Review by cpcundill — APPROVED** [review]

**Pull request merged**